validation/src/test/scala/hmda/validation/rules/ts/validity/V155Spec.scala

```diff
@@ -12,15 +12,17 @@ class V155Spec extends PropSpec with PropertyChecks with MustMatchers with TsGen
   property("A valid email address for the institution must be provided") {
     forAll(tsGen) { ts =>
       whenever(ts.contact.email != "") {
-        V155(ts.contact) mustBe Success()
+        V155(ts) mustBe Success()
       }
     }
   }
 
   property("Invalid emails should be detected") {
-    forAll(badContactGen) { contact =>
-      whenever(contact.email != "") {
-        V155(contact) mustBe Failure("is not a valid email")
+    forAll(tsGen) { ts =>
+      val badContact = badContactGen.sample.getOrElse(Contact("", "", "", ""))
+      val badTs = ts.copy(contact = badContact)
+      whenever(badTs.contact.email != "") {
+        V155(badTs) mustBe Failure("is not a valid email")
```

Comment:
Yeah, that's exactly what happened: I ran the tests and it didn't happen. (Later, I ran `sbt test` 10 times, and it happened in 2 of those times.)  It seem, then, like property-based testing (or at least this particular test) is not the right way to test this rule (or specifically, this behavior).  The empty-string case is a known, expected case, and I'd want it to be covered every time I run the tests.  Otherwise, I can't be confident that running the tests lets me know I haven't broken the code.

On the plus side, that explains why `sample` has the signature it has! I'd wondered, since I had literally never seen it return `None` when I was using it to test (well, visually inspect) generators in the console.